### DISCUSSION TITLE
Using list aggregator in groupby

### DISCUSSION THREAD

**Discussion by vicw** [vaexio/vaex#2027] [Q&A]
Hi,

I recently came across the list aggregator and thought it would allow me to apply a custom function to each group, but am having trouble getting the code to work.  I am trying to calculate the first differences within each group, returning a list that is the same shape as the input, using the following code:

```
x = vaex.from_arrays(x=np.asarray(list("aaaabb")),y=np.arange(6)).groupby('x').agg({'y':'list'})
x['z'] = x.y.apply(lambda x:np.concatenate(([x[0]],np.diff(x))))
```

Also, after applying the function on the list, is there a way to invert the transformation, and "stack" the list back into the original shape?

Thanks for any help in advance.

**Comment by maartenbreddels** [comment]
Hi,

No, that is not possible, but would be interesting.
I wonder if it wouldn't be simpler to sort the dataframe by 'x', then filter it in as many dataframes as you have 'x' values. Then apply the diff, and concatenate them together again.

Regards,

Maarten
